UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -295,18 +275,6 @@ public int GetStepCount()
             return m_StepCount;
         }
 
-        /// <summary>
-        /// Resets the step reward and possibly the episode reward for the agent.
-        /// </summary>
-        public void ResetReward()
```

Comment:
The places were this was removed don't account for the case where `m_Done` is `true`

Reply:
The reset logic is really confusing.  We have:
- `ResetIfDone`
- `_AgentReset`
- `AgentReset`
- `ResetData`
- `ForceReset`

Just by looking, I have no idea which is done when or why. 
 
This issue may not be in the scope of PR, but it feels like something that should be addressed.

Reply:
They do, I reset the Cumulative reward in ResetData

Reply:
and `ResetReward`